### DISCUSSION TITLE
Remove distutils.cfg from Python forumlas

### DISCUSSION THREAD

**Discussion by uranusjr** [Homebrew/discussions#1405] [Tap maintenance and Homebrew development] [answered]
Not sure where best this should go, I’m entirely green to Homebrew. Sorry if I mess up things.

I am one of the maintainers of pip, the Python package installer. We are recently informed that Homebrew is currently using `distutils.cfg` to configure package installation schemes, and we believe it is best for Homebrew to remove the usage.

Python is currently deprecating distutils, and will remove the module entirely in 3.10 ([PEP 632](https://www.python.org/dev/peps/pep-0632/)). As a part of the transition, pip is moving all its current `distutils` usages to the `sysconfig` module instead, and a user recently reported pip would no longer work correctly with Homebrew-compiled Python after the transision. This is due to Homebrew’s usage of `distutils.cfg`, which is no longer supported in `sysconfig`.

The `distutils.cfg` file (and its various variants) has long been disfavoured by Python core maintainers, and the recommendation, if redistributors wish to provide custom install schemes, has been to directly patch `distutils` and `sysconfig` instead. This is the current approach taken by all Linux distributions that we have contact with, and the community is working toward providing better support to the patching process ([b.p.o-43976](https://bugs.python.org/issue43976)). I’m recommending Homebrew to also migrate away from the `distutils.cfg` usage, since it is going to break while the community continuously migrate away from  `distutils`, and stop working entirely when Python 3.12 lands, and join the conversation with other Python redistributors to improve `sysconfig` to support Homebrew’s needs.

**Comment by carlocab** [comment, marked as answer]
Thanks, @uranusjr. I've opened https://github.com/Homebrew/homebrew-core/issues/76621 to track this properly.

**Comment by SMillerDev** [comment]
Please make a pull request with your suggested fix and I'll gladly review it.

**Reply by uranusjr** [reply]
I would love to except I don’t actually have Homebrew installed anywhere 🙂 I’ll try to create an installation and see if I can come up with something. But if anyone else wants to do it, I can tell you exactly what lines to patch as well.
